File: Medium/1657. Determine if Two Strings Are Close/Solution.py
from collections import Counter

class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        freq1 = Counter(word1)
        freq2 = Counter(word2)
        
        # Check if they have the same unique characters
        if set(freq1.keys()) != set(freq2.keys()):
            return False
        
        # Check if frequency distributions are the same
        return sorted(freq1.values()) == sorted(freq2.values())

File: Medium/1657. Determine if Two Strings Are Close/Solution.cs
using System;

public class Solution {
    public bool CloseStrings(string word1, string word2) {
        int[] freq1 = new int[26];
        int[] freq2 = new int[26];

        foreach (char c in word1) {
            freq1[c - 'a']++;
        }

        foreach (char c in word2) {
            freq2[c - 'a']++;
        }

        for (int i = 0; i < 26; i++) {
            if ((freq1[i] > 0 && freq2[i] == 0) || (freq2[i] > 0 && freq1[i] == 0)) {
                return false;
            }
        }

        Array.Sort(freq1);
        Array.Sort(freq2);

        for (int i = 0; i < 26; i++) {
            if (freq1[i] != freq2[i]) {
                return false;
            }
        }
        return true;
    }
}